packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,408 @@
+      this._onDecrement = props.onDecrement;
+    }
+
+    this.focus = this.focus.bind(this);
+  }
+
+  /**
+  * Invoked when a component is receiving new props. This method is not called for the initial render.
+  */
+  public componentWillReceiveProps(newProps: ISpinButtonProps): void {
+    this._lastValidValue = this.state.value;
+    let value: string = newProps.value ? newProps.value : String(newProps.min);
+    if (newProps.defaultValue) {
+      value = String(Math.max(newProps.min, Math.min(newProps.max, Number(newProps.defaultValue))));
+    }
+
+    this.setState({
+      value: value
+    });
+  }
+
+  public render() {
+    const {
+      disabled,
+      label,
+      min,
+      max,
+      width: spinbuttonWidth,
+      labelPosition,
+      iconProps,
+      incrementButtonIcon,
+      decrementButtonIcon
+    } = this.props;
+
+    const {
+      value,
+      keyboardSpinDirection
+    } = this.state;
+
+    return (
+      <div className={ styles.SpinButtonContainer } style={ spinbuttonWidth && { width: spinbuttonWidth } }>
+        { labelPosition !== Position.bottom && <div className={ styles.labelWrapper } style={ this._labelDirectionHelper() }>
+          { iconProps && <Icon iconName={ iconProps.iconName } className={ css(styles.SpinButtonIcon) } aria-hidden='true'></Icon> }
+          { label &&
+            < Label
+              id={ this._labelId }
+              htmlFor={ this._inputId }
+              className={ styles.SpinButtonLabel }>{ label }
+            </Label>
+          }
+        </div> }
+        <div className={ css(styles.SpinButtonWrapper, ((labelPosition === Position.top || labelPosition === Position.bottom) ? styles.topBottom : '')) }>
+          < input
+            value={ value }
+            id={ this._inputId }
+            onChange={ this._onChange }
+            onInput={ this._onInputChange }
+            className={ css(styles.Input, (disabled ? styles.disabled : '')) }
+            type='text'
+            role='spinbutton'
+            aria-labelledby={ label && this._labelId }
+            aria-valuenow={ value }
+            aria-valuemin={ min && String(min) }
+            aria-valuemax={ max && String(max) }
+            onBlur={ this._validate }
+            ref={ this._resolveRef('_input') }
+            onFocus={ this.focus }
+            onKeyDown={ this._handleKeyDown }
+            onKeyUp={ this._handleKeyUp }
+            readOnly={ disabled }
+            aria-disabled={ disabled }
+          />
+          <span className={ styles.ArrowBox }>
+            <IconButton
+              className={ css('ms-UpButton', styles.UpButton, (keyboardSpinDirection === KeyboardSpinDirection.up ? styles.active : '')) }
+              disabled={ disabled }
+              iconProps={ incrementButtonIcon }
+              title='Increase'
+              aria-hidden='true'
+              onMouseDown={ () => { this._updateValue(true /* shouldSpin */, this._onIncrement); } }
```

Comment:
As is now, every time it renders is creating a new function that is bind to onMouseDown. If you use a a class method or store it in a field it would not. This typically breaks pure components since you pass a new reference hence shallow reference checks fail.